### PR TITLE
Screen fields and more

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
The fields in the `Screen` subclasses are really important to SEI

Changelog:-
```css
 On branch dev6
 Changes to be committed:
	renamed:    mappings/net/minecraft/client/gui/screen/StatsListener.mapping -> mappings/net/minecraft/client/gui/StatsListener.mapping
	modified:   mappings/net/minecraft/client/gui/screen/CommandBlockScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/ConfirmChatLinkScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/CustomizeFlatLevelScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/CustomizeWorldScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/CustomizedWorldPresetsScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/DemoScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/DirectConnectScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/DisconnectedScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/DownloadingTerrainScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/FatalErrorScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/GameMenuScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/OpenToLanScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/PresetsScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/ResourcePackScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/SoundsScreen.mapping
	deleted:    mappings/net/minecraft/client/gui/screen/ingame/AbstractInventoryScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/ingame/BeaconScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/ingame/BrewingStandScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/ingame/CraftingTableScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/ingame/InventoryScreen.mapping
	new file:   mappings/net/minecraft/client/gui/screen/ingame/SurvivalInventoryScreen.mapping
	modified:   mappings/net/minecraft/client/sound/MusicTracker.mapping
	modified:   mappings/net/minecraft/util/PacketByteBuf.mapping
	modified:   mappings/net/minecraft/util/ProgressListener.mapping
	renamed:    mappings/net/minecraft/class_417.mapping -> mappings/net/minecraft/world/gen/CustomizedWorldProperties.mapping
	new file:   mappings/net/minecraft/world/gen/FlatWorldHelper.mapping
	new file:   mappings/net/minecraft/world/gen/layer/FlatWorldLayer.mapping
	new file:   mappings/net/minecraft/client/gui/CreativeInventoryListener.mapping
	new file:   mappings/net/minecraft/client/gui/EnchantingPhrases.mapping
	renamed:    mappings/net/minecraft/client/gui/screen/ingame/GenericContainerScreen.mapping -> mappings/net/minecraft/client/gui/screen/ingame/ChestScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/ingame/CraftingTableScreen.mapping
	deleted:    mappings/net/minecraft/client/gui/screen/ingame/CreativeInventoryListener.mapping
	new file:   mappings/net/minecraft/client/gui/screen/ingame/DispenserScreen.mapping
	deleted:    mappings/net/minecraft/client/gui/screen/ingame/EnchantingPhrases.mapping
	deleted:    mappings/net/minecraft/client/gui/screen/ingame/Generic3x3ContainerScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/ingame/HorseScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/ingame/VillagerTradingScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/multiplayer/MultiplayerScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/options/ControlsOptionsScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/recipebook/RecipeBookGhostSlots.mapping
	modified:   mappings/net/minecraft/client/gui/screen/resourcepack/ResourcePackListWidget.mapping
	new file:   mappings/net/minecraft/client/gui/screen/resourcepack/ResourcePackWidget.mapping
	modified:   mappings/net/minecraft/client/gui/screen/world/CreateWorldScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/world/EditWorldScreen.mapping
	modified:   mappings/net/minecraft/client/gui/screen/world/SelectWorldScreen.mapping
	new file:   mappings/net/minecraft/client/gui/widget/LanServerEntry.mapping
	modified:   mappings/net/minecraft/client/gui/widget/ListWidget.mapping
	new file:   mappings/net/minecraft/client/gui/widget/ServerEntry.mapping
	modified:   mappings/net/minecraft/client/render/GuiLighting.mapping
	modified:   mappings/net/minecraft/client/render/entity/PlayerModelPart.mapping
	modified:   mappings/net/minecraft/container/ContainerListener.mapping
	modified:   mappings/net/minecraft/client/gui/screen/ingame/ChestScreen.mapping
	modified:   mappings/net/minecraft/util/PacketByteBuf.mapping
	modified:   mappings/net/minecraft/block/AnvilBlock.mapping
	modified:   mappings/net/minecraft/item/Item.mapping
	modified:   mappings/net/minecraft/item/ItemStack.mapping
	renamed:    mappings/net/minecraft/world/biome/layer/type/NameableContainerProvider.mapping -> mappings/net/minecraft/world/biome/layer/type/Nameable.mapping
	deleted:    mappings/net/minecraft/class_1581.mapping
	deleted:    mappings/net/minecraft/class_1629.mapping
	modified:   mappings/net/minecraft/command/AbstractCommand.mapping
	modified:   mappings/net/minecraft/command/CommandStats.mapping
	new file:   mappings/net/minecraft/server/command/CommandRegistry.mapping
	new file:   mappings/net/minecraft/server/command/CommandRegistryProvider.mapping
	deleted:    mappings/net/minecraft/class_428.mapping
	modified:   mappings/net/minecraft/server/MinecraftServer.mapping
	modified:   mappings/net/minecraft/server/dedicated/DedicatedServer.mapping
	new file:   mappings/net/minecraft/server/dedicated/DedicatedServerProvider.mapping
	deleted:    mappings/net/minecraft/server/dedicated/MinecraftDedicatedServer.mapping
	new file:   mappings/net/minecraft/world/chunk/EndChunkGenerator.mapping
	new file:   mappings/net/minecraft/world/gen/NoiseGenerator.mapping
```

### CODE REVIEW

**Comment by hYdos** [comment]
oh forgot you can request reviews now

**Comment by BoogieMonster1O1** [comment]
> oh forgot you can request reviews now

I can, and i can also add labels

File: mappings/net/minecraft/client/gui/screen/DirectConnectScreen.mapping [hunk 1]
```diff
@@ -1 +1,7 @@
 CLASS net/minecraft/class_738 net/minecraft/client/gui/screen/DirectConnectScreen
+	FIELD field_3144 parent Lnet/minecraft/class_754;
+	FIELD field_3145 info Lnet/minecraft/class_910;
+	FIELD field_3146 search Lnet/minecraft/class_689;
```
Comment (shedaniel) [inline mappings/net/minecraft/client/gui/screen/DirectConnectScreen.mapping:4]: Would want to suffix all these so it becomes `searchField`

File: mappings/net/minecraft/client/gui/screen/DirectConnectScreen.mapping [hunk 2]
```diff
@@ -1 +1,7 @@
 CLASS net/minecraft/class_738 net/minecraft/client/gui/screen/DirectConnectScreen
+	FIELD field_3144 parent Lnet/minecraft/class_754;
+	FIELD field_3145 info Lnet/minecraft/class_910;
```
Comment (shedaniel) [inline mappings/net/minecraft/client/gui/screen/DirectConnectScreen.mapping:3]: `serverInfo` or `server`?

File: mappings/net/minecraft/client/gui/screen/ingame/InventoryScreen.mapping [hunk 3]
```diff
@@ -1,12 +1,4 @@
...
-	METHOD <init> (Lnet/minecraft/class_1963;)V
-		ARG 1 player
-	METHOD method_2946 renderEntity (IIIFFLnet/minecraft/class_1752;)V
-		ARG 0 x
-		ARG 1 y
-		ARG 2 size
-		ARG 3 mouseX
-		ARG 4 mouseY
-		ARG 5 entity
+CLASS net/minecraft/class_799 net/minecraft/client/gui/screen/ingame/InventoryScreen
```
Comment (shedaniel) [inline mappings/net/minecraft/client/gui/screen/ingame/InventoryScreen.mapping:1]: Are you sure that this is the correct name? I would name this `AbstractInventoryScreen` as in old official yarn.

File: mappings/net/minecraft/client/gui/screen/ingame/SurvivalInventoryScreen.mapping [hunk 4]
```diff
@@ -0,0 +1,12 @@
+CLASS net/minecraft/class_806 net/minecraft/client/gui/screen/ingame/SurvivalInventoryScreen
```
Comment (shedaniel) [inline mappings/net/minecraft/client/gui/screen/ingame/SurvivalInventoryScreen.mapping:1]: `InventoryScreen` for official old yarn

File: mappings/net/minecraft/client/gui/screen/recipebook/RecipeBookGhostSlots.mapping [hunk 5]
```diff
@@ -1,2 +1,3 @@
 CLASS net/minecraft/class_117 net/minecraft/client/gui/screen/recipebook/RecipeBookGhostSlots
+	FIELD field_442 source Lnet/minecraft/class_123;
```
Comment (shedaniel) [inline mappings/net/minecraft/client/gui/screen/recipebook/RecipeBookGhostSlots.mapping:2]: `recipe`?

File: mappings/net/minecraft/client/gui/screen/ingame/BeaconScreen.mapping [hunk 6]
```diff
@@ -1,4 +1,8 @@
 CLASS net/minecraft/class_783 net/minecraft/client/gui/screen/ingame/BeaconScreen
+	FIELD field_3408 LOGGER Lorg/apache/logging/log4j/Logger;
+	FIELD field_3409 beaconTexture Lnet/minecraft/class_1605;
```
Comment (shedaniel) [inline mappings/net/minecraft/client/gui/screen/ingame/BeaconScreen.mapping:3]: did mojang concern and make this not static final

File: mappings/net/minecraft/client/gui/screen/DirectConnectScreen.mapping (hunk 2 above)
Comment (BoogieMonster1O1) [reply mappings/net/minecraft/client/gui/screen/DirectConnectScreen.mapping:3]: Should I rename it to `serverInfo`?

File: mappings/net/minecraft/client/gui/screen/DirectConnectScreen.mapping (hunk 1 above)
Comment (BoogieMonster1O1) [reply mappings/net/minecraft/client/gui/screen/DirectConnectScreen.mapping:4]: Sure I'll do that a few hours later

File: mappings/net/minecraft/client/gui/screen/ingame/BeaconScreen.mapping (hunk 6 above)
Comment (BoogieMonster1O1) [reply mappings/net/minecraft/client/gui/screen/ingame/BeaconScreen.mapping:3]: I'll look into that

**Pull request merged**